TYPO3.Flow/Classes/TYPO3/Flow/Http/Component/TrustedProxiesComponent.php

```diff
@@ -50,14 +50,21 @@ public function handle(ComponentContext $componentContext)
         }
 
         $hostHeader = $this->getFirstTrustedProxyHeaderValue(self::HEADER_HOST, $trustedRequest);
+        $portFromHost = null;
         if ($hostHeader !== null) {
+            $portSeparator = strrpos($hostHeader, ':');
+            if ($portSeparator > 0) {
```

Comment:
strrpos returns `FALSE` if the needle doesn't exist in the haystack string. Also &lt;nitpick&gt;that variable won't contain the port separator but it's position&lt;/nitpick&gt;
I suggest the following code:
```php
list($hostHeaderWithoutPort, $portFromHost) = explode(':', $hostHeader);
$portSeparator = strrpos($hostHeader, ':');
if ($portSeparator !== false) {
    $trustedRequest->getUri()->setPort($portFromHost);
}
$trustedRequest->getUri()->setHost($hostHeaderWithoutPort);
```
That would also allow a host of ":<port>" (even though I'm not sure if that's valid tbh).
If you agree I can push that as follow-up with a corresponding test